src/history/hs_rec.c

```diff
@@ -274,6 +274,8 @@ __wt_hs_insert_updates(
     WT_DECL_ITEM(prev_full_value);
     WT_DECL_ITEM(tmp);
     WT_DECL_RET;
+/* Limit the number of consecutive reverse modifies. */
+#define WT_MAX_CONSECUTIVE_REVERSE_MODIFY 10
```

Comment:
Is this number chosen randomly?

Reply:
As discussed in https://jira.mongodb.org/browse/WT-7106, it has been decided after a set of tests that showed the trade off between history store size and read latency. A higher number would potentially decrease the size of the history store but then have an impact on the read latency. It has been decided to start with 10 but it could be tweaked if required.

Reply:
Got it